stl/base.py

```diff
@@ -89,6 +89,9 @@ def __init__(self, name):
   def __eq__(self, other):
     return NamedObject.__eq__(self, other) and self.params == other.params
 
+  def Resolve(self, env, resolved_params):
+      raise NotImplementedError('Resolved() is not needed: ' + self.name)
+
```

Comment:
What about:

if True:
  raise NotImplementedError('Resolved() is not needed: ' + self.name)

in base class? Would "if True" do some trick?